fuel-vm/src/tests/upload.rs

```diff
@@ -0,0 +1,366 @@
+#![allow(non_snake_case)]
+use crate::{
+    checked_transaction::IntoChecked,
+    interpreter::Interpreter,
+    storage::UploadedBytecode,
+};
+use fuel_asm::{
+    op,
+    PanicReason,
+};
+use fuel_tx::{
+    field::Outputs,
+    policies::Policies,
+    GasCosts,
+    Input,
+    Output,
+    Transaction,
+    Upload,
+    UploadPart,
+    ValidityError,
+};
+use fuel_types::AssetId;
+
+use crate::{
+    checked_transaction::Ready,
+    storage::UploadedBytecodes,
+};
+use fuel_storage::StorageAsRef;
+
+use crate::{
+    checked_transaction::CheckError,
+    error::InterpreterError,
+};
+#[cfg(feature = "alloc")]
+use alloc::{
+    vec,
+    vec::Vec,
+};
+
+const AMOUNT: u64 = 1000;
+const BYTECODE_SIZE: usize = 1024;
+
+fn bytecode() -> Vec<u8> {
+    vec![123; BYTECODE_SIZE]
+}
+
+fn valid_input() -> Input {
+    let predicate = vec![op::ret(1)].into_iter().collect::<Vec<u8>>();
+    let owner = Input::predicate_owner(&predicate);
+    Input::coin_predicate(
+        Default::default(),
+        owner,
+        AMOUNT,
+        AssetId::BASE,
+        Default::default(),
+        Default::default(),
+        predicate,
+        vec![],
+    )
+}
+
+fn valid_transaction_from_part(part: UploadPart) -> Ready<Upload> {
+    Transaction::upload_from_part(
+        part,
+        Policies::new().with_max_fee(AMOUNT),
+        vec![valid_input()],
+        vec![Output::change(Default::default(), 0, AssetId::BASE)],
+        vec![],
+    )
+    .into_checked_basic(Default::default(), &Default::default())
+    .expect("Failed to generate checked tx")
+    .test_into_ready()
+}
+
+#[test]
+fn transact_uploads_bytecode_with_one_part() {
```

Comment:
These tests are beautiful. Thanks!

Nit: I prefer to put double `__` between the SUT and the test case: `transact__uploads_bytecode_with_one_part`. If you don't like that then no pressure, it's not industry standard, but we already have a `#![allow(non_snake_case)]` on the file, so it wouldn't hurt.

Regardless, I don't want to be too picky since (like I said) these tests are already above par and the test name would be semantically identical.